src/app/services/favorite.service.ts

```diff
@@ -28,20 +30,35 @@ export class FavoriteService {
   }
 
   insert(custId: number, prodId: number): Observable<any> {
-    return this.http.post(`${environment.apiBase}/favorite`,
-    {userId: custId, productId: prodId})
+    let fave = new Favorite();
+    let cust = new User();
+    let product = new Product();
+    cust.userId = custId;
+    product.productId = prodId;
+    fave.user = cust;
+    fave.product = product;
+    return this.http.post(`${environment.apiBase}/favorite`, fave)
     .pipe( map( (response: any) => response));
   }
 
   update(favoriteId: number, custId: number, prodId: number): Observable<any> {
-    return this.http.put(`${environment.apiBase}/favorite`,
-    {favoriteId: favoriteId, userId: custId, productId: prodId})
+    let fave = new Favorite();
```

Comment:
can reduce this code to
const fave = new Favorite(favoriteId: favoriteId, user: cust, product: product);

Reply:
verbosity reduced